src/multivariate/multinomial.jl

```diff
@@ -118,16 +122,18 @@ end
 function _logpdf{T<:Real}(d::Multinomial, x::AbstractVector{T})
     p = probs(d)
     n = ntrials(d)
-    s = lgamma(n + 1.0)
-    t = 0
+    S = eltype(p)
+    R = promote_type(T, S)
+    s = R(lgamma(n + 1))
```

Comment:
Couldn't `R` end up being `Int` if both `T` and `S` are `Int`s. The this would throw an inexact error.